### PR TITLE
[SDPA-3045] Added request id in all route requests

### PR DESCRIPTION
**Pull request by tim-yao**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-3045

### Changed

1.  Added a unique id for each route request
2. Added the request id in BE request header
3. Added the request id in log
4. Rename "results" to "pageData" in middleware to make it meaningful.

Screen Shot 2019-08-22 at 9 23 13 am

### CODE REVIEW

File: packages/ripple-nuxt-tide/lib/core/middleware.js [hunk 1]
```diff
@@ -4,9 +4,10 @@ import { isPreviewPath } from '../../modules/authenticated-content/lib/preview'
 import logger from './logger'
 
 // Fetch page data from Tide API by current path
-export default async function (context, results) {
-  results.tidePage = null
-  results.tideErrorType = null
+export default async function (context, pageData) {
+  pageData.tidePage = null
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/core/middleware.js]: I renamed `results` to `pageData` for middleware as I have done in previous middleware PR, I think it more make sense to me.

File: packages/ripple-nuxt-tide/lib/core/tide-helper.js [hunk 2]
```diff
@@ -1,6 +1,11 @@
-
+import cuid from 'cuid'
 import mime from 'mime-types'
 
+// Generate a unique id
+export const generateId = () => {
+  return cuid()
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/core/tide-helper.js]: Use cuid to generate a unique id for use.

File: packages/ripple-nuxt-tide/lib/server-middleware/request-id.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Log server connection
+import { generateId } from './../core/tide-helper'
+const url = require('url')
+
+module.exports = function (req, res, next) {
+  // req is the Node.js http request object
+  const reqUrl = decodeURI((url.parse(req.url)).pathname)
+  if (!reqUrl.includes('/api/v')) {
+    req.requestId = generateId()
+  }
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/server-middleware/request-id.js]: Added a server middleware to generate id on every server request.

File: packages/ripple-nuxt-tide/lib/templates/request-id.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { generateId } from '@dpc-sdp/ripple-nuxt-tide/lib/core/tide-helper'
+import middleware from './middleware'
+
+middleware['request-id'] = async (context) => {
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/templates/request-id.js]: Added a route middleware for generate request id if it's client side navigation.

File: packages/ripple-nuxt-tide/lib/core/tide.js [hunk 5]
```diff
@@ -19,9 +19,9 @@ export const tide = (axios, site, config) => ({
    * @param {String} resource Resource type e.g. <entity type>/<bundle>
    * @param {Object} params Object to convert to QueryString. Passed in URL.
    * @param {String} id Resource UUID
-   * @param {String} authToken Authentication token
+   * @param {Object} headersConfig Object of headers config
```
Comment (dylankelly) [inline packages/ripple-nuxt-tide/lib/core/tide.js]: not a very good description, what is it used for? Something like `axios request headers config object` might be better

**Pull request merged**